utils/kubernetes/kube_state_processor.py

```diff
@@ -20,13 +25,38 @@ def __init__(self, kubernetes_check):
             # 'unknown':   AgentCheck.UNKNOWN
         }
 
+        # these metrics will be extracted with all their labels and reported as-is with their corresponding metric name
+        self.metric_to_gauge = {
+            # message.metric: datadog metric name
+            'kube_node_status_capacity_cpu_cores': NAMESPACE + '.node.cpu_capacity',
+            'kube_node_status_capacity_memory_bytes': NAMESPACE + '.node.memory_capacity',
+            'kube_node_status_capacity_pods': NAMESPACE + '.node.pods_capacity',
+            'kube_node_status_allocatable_cpu_cores': NAMESPACE + '.node.cpu_allocatable',
+            'kube_node_status_allocatable_memory_bytes': NAMESPACE + '.node.memory_allocatable',
+            'kube_node_status_allocatable_pods': NAMESPACE + '.node.pods_allocatable',
+            'kube_deployment_status_replicas_available': NAMESPACE + '.deployment.replicas_available',
+            'kube_deployment_status_replicas_unavailable': NAMESPACE + '.deployment.replicas_unavailable',
+            'kube_deployment_status_replicas_updated': NAMESPACE + '.deployment.replicas_updated',
+            'kube_deployment_spec_replicas': NAMESPACE + '.deployment.replicas_desired',
+            'kube_pod_container_resource_requests_cpu_cores': NAMESPACE + '.container.cpu_requested',
+            'kube_pod_container_resource_requests_memory_bytes': NAMESPACE + '.container.memory_requested',
+            'kube_pod_container_resource_limits_cpu_cores': NAMESPACE + '.container.cpu_limit',
+            'kube_pod_container_resource_limits_memory_bytes': NAMESPACE + '.container.memory_limit',
+            'kube_pod_container_status_restarts': NAMESPACE + '.container.restarts'
+        }
+
     def process(self, message, **kwargs):
         """
-        Search this class for a method with the same name of the message and
-        invoke it. Log some info if method was not found.
+        Handle a message according to the following flow:
+            - search self.metric_to_gauge for a prometheus.metric <--> datadog.metric mapping
+            - call check method with the same name as the metric
+            - log some info if none of the above worked
         """
         try:
-            getattr(self, message.name)(message, **kwargs)
+            if message.name in self.metric_to_gauge:
+                self._submit_gauges(self.metric_to_gauge[message.name], message)
```

Comment:
i feel like it could be useful someday, but yeah let's kill it for now since it's not used.